components/datetime/src/date.rs

```diff
@@ -189,3 +190,79 @@ impl<'data, T: ZonedDateTimeInput> LocalizedDateTimeInput<T>
         todo!("#487")
     }
 }
+
+impl<A: AsCalendar<Calendar = Gregorian>> DateInput for Date<A> {
```

Comment:
Issue: I think we should be able to implement this trait agnostic of `A`.  Otherwise, how does this scale to many more calendars?  What can we do to make it so that `impl<A> DateInput for Date<A>` works?

Reply:
We should! That's not the purpose of this PR, this PR is trying to hook stuff together without adding support for other calendars. I didn't go further because we only have ISO and Gregorian, and ISO doesn't have actual data. We can support ISO as a hack over Gregorian data loading but I'd rather pick a different calendar and hook it in so that we can test it.

To make that impl work we need `DateInput` to have some idea of what data key needs to be accessed. We likely need to do some work on era codes/etc as well, and a CLDR transformer for eras. It's a bunch of stuff that is not necessary for this PR (and I don't want to bundle too many things together)

I could absolutely implement this trait agnostic of `A` in this PR itself but it would be incorrect: `Date<Iso>` would be formattable as a Gregorian date (as would any other dates being written). I didn't want the API to pretend to support things it doesn't, I'd rather expand this when we add the necessary support (and that way we can test it properly)